### PR TITLE
Mention one-to-one mapping of PVCs and PVs in the docs and examples

### PR DESCRIPTION
**Pull request by unexge**
The Driver only supports Static Provisioning as of today, and in order to configure it correctly, one needs to ensure there is no `storageClassName` set (it should be `""`) and also there is a one-to-one mapping between PVC and PV using `claimRef`. This PR clarifies this requirement and updates example manifests to specify one-to-one mapping between PVC and PV. See a recent issue regarding this: https://github.com/awslabs/mountpoint-s3-csi-driver/issues/249

--- 
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/CONFIGURATION.md [hunk 1]
```diff
@@ -14,15 +48,15 @@ Mountpoint CSI driver. For more details on creating a policy and an IAM role, re
 ["Creating an IAM role"](https://docs.aws.amazon.com/eks/latest/userguide/s3-csi.html#s3-create-iam-role) from the
 EKS User Guide.
 
-The Mountpoint CSI Driver can be configured to ingest credentials via two approaches: globally for the entire 
+The Mountpoint CSI Driver can be configured to ingest credentials via two approaches: globally for the entire
```
Comment (muddyfish) [inline docs/CONFIGURATION.md:51]: Nit: Can these single character changes go in their own PR so it doesn't confuse the commit history?

**Review by muddyfish — APPROVED** [review]

File: docs/CONFIGURATION.md (hunk 1 above)
Comment (unexge) [reply docs/CONFIGURATION.md:51]: I think that should be fine as this is a doc-only change. For long-term, we should probably look for enforcing a standard format for Go (`gofmt`), YAML and Markdown (`Prettier`).

Comment (dannycjones) [reply docs/CONFIGURATION.md:51]: I agree with Simon, if we do find any changes like this then let's just make two commits and merge them in separately.

Agree enforcing format checkers would be better, but I'd prefer we remove the noise from PRs and commit history regardless.
